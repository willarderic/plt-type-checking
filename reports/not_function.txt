TOKEN: LPAREN  LEXEME: (					TOKEN: LPAREN    LEXEME: (
TOKEN: NOT  LEXEME: not						TOKEN: NOT       LEXEME: not
UN_OP -> not							UN_OP -> not
TOKEN: LPAREN  LEXEME: (					TOKEN: LPAREN    LEXEME: (
TOKEN: LAMBDA  LEXEME: lambda					TOKEN: LAMBDA    LEXEME: lambda

___Entering new scope... 					___Entering new scope...

TOKEN: LPAREN  LEXEME: (					TOKEN: LPAREN    LEXEME: (
ID_LIST -> epsilon						ID_LIST -> epsilon
TOKEN: IDENT  LEXEME: x						TOKEN: IDENT     LEXEME: x
ID_LIST -> ID_LIST IDENT					ID_LIST -> ID_LIST IDENT
___Adding x to symbol table					___Adding x to symbol table
TOKEN: IDENT  LEXEME: y						TOKEN: IDENT     LEXEME: y
ID_LIST -> ID_LIST IDENT					ID_LIST -> ID_LIST IDENT
___Adding y to symbol table					___Adding y to symbol table
TOKEN: RPAREN  LEXEME: )					TOKEN: RPAREN    LEXEME: )
TOKEN: LPAREN  LEXEME: (					TOKEN: LPAREN    LEXEME: (
TOKEN: ADD  LEXEME: +						TOKEN: ADD       LEXEME: +
ARITH_OP -> +							ARITH_OP -> +
BIN_OP -> ARITH_OP						BIN_OP -> ARITH_OP
TOKEN: IDENT  LEXEME: x						TOKEN: IDENT     LEXEME: x
EXPR -> IDENT							EXPR -> IDENT
TOKEN: IDENT  LEXEME: y						TOKEN: IDENT     LEXEME: y
EXPR -> IDENT							EXPR -> IDENT
ARITHLOGIC_EXPR -> BIN_OP EXPR EXPR				ARITHLOGIC_EXPR -> BIN_OP EXPR EXPR
PARENTHESIZED_EXPR -> ARITHLOGIC_EXPR				PARENTHESIZED_EXPR -> ARITHLOGIC_EXPR
TOKEN: RPAREN  LEXEME: )					TOKEN: RPAREN    LEXEME: )
EXPR -> ( PARENTHESIZED_EXPR )					EXPR -> ( PARENTHESIZED_EXPR )
LAMBDA_EXPR -> lambda ( ID_LIST ) EXPR				LAMBDA_EXPR -> lambda ( ID_LIST ) EXPR

___Exiting scope...						___Exiting scope...

PARENTHESIZED_EXPR -> LAMBDA_EXPR				PARENTHESIZED_EXPR -> LAMBDA_EXPR
TOKEN: RPAREN  LEXEME: )					TOKEN: RPAREN    LEXEME: )
EXPR -> ( PARENTHESIZED_EXPR )					EXPR -> ( PARENTHESIZED_EXPR )
ARITHLOGIC_EXPR -> UN_OP EXPR					ARITHLOGIC_EXPR -> UN_OP EXPR
Line 1: Arg 1 cannot be function				Line 1: Arg 1 cannot be function
