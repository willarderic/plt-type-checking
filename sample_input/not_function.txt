(not (lambda (x y) (+ x y)))


