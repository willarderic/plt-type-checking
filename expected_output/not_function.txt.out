TOKEN: LPAREN    LEXEME: (
TOKEN: NOT       LEXEME: not
UN_OP -> not
TOKEN: LPAREN    LEXEME: (
TOKEN: LAMBDA    LEXEME: lambda

___Entering new scope...

TOKEN: LPAREN    LEXEME: (
ID_LIST -> epsilon
TOKEN: IDENT     LEXEME: x
ID_LIST -> ID_LIST IDENT
___Adding x to symbol table
TOKEN: IDENT     LEXEME: y
ID_LIST -> ID_LIST IDENT
___Adding y to symbol table
TOKEN: RPAREN    LEXEME: )
TOKEN: LPAREN    LEXEME: (
TOKEN: ADD       LEXEME: +
ARITH_OP -> +
BIN_OP -> ARITH_OP
TOKEN: IDENT     LEXEME: x
EXPR -> IDENT
TOKEN: IDENT     LEXEME: y
EXPR -> IDENT
ARITHLOGIC_EXPR -> BIN_OP EXPR EXPR
PARENTHESIZED_EXPR -> ARITHLOGIC_EXPR
TOKEN: RPAREN    LEXEME: )
EXPR -> ( PARENTHESIZED_EXPR )
LAMBDA_EXPR -> lambda ( ID_LIST ) EXPR

___Exiting scope...

PARENTHESIZED_EXPR -> LAMBDA_EXPR
TOKEN: RPAREN    LEXEME: )
EXPR -> ( PARENTHESIZED_EXPR )
ARITHLOGIC_EXPR -> UN_OP EXPR
Line 1: Arg 1 cannot be function
